### DISCUSSION TITLE
Reset password not working

### DISCUSSION THREAD

**Discussion by SmartManoj** [fastapi-users/fastapi-users#1017] [Q&A] [answered]
## Describe the bug

![image](https://user-images.githubusercontent.com/7231077/174231310-b7c8d084-d30e-489a-92a1-6aa8f943ee26.png)

To be filled

## Configuration
 - Python version : 3.10.4
 - FastAPI version : 0.65.2
 - FastAPI Users version : 10.0.7

### FastAPI Users configuration

```py
# Please copy/paste your FastAPI Users configuration here.
```

**Comment by frankie567** [comment]
Could you give more details about the steps you are following and the configuration you have? A screenshot is not sufficient to help you.

From what I see, the token is probably expired.

**Comment by SmartManoj** [comment]
Tried this example. https://github.com/fastapi-users/fastapi-users/tree/master/examples/sqlalchemy

**Comment by frankie567** [comment, marked as answer]
Ok, so what did you do? How did you generate the reset token?

**Reply by SmartManoj** [reply]
thru /forgot_password route

**Reply by frankie567** [reply]
A few things to check then:

- Are you sure you copy-pasted the token correctly?
- Has the token been generated less than an hour ago?
- Does the user still exist in DB?

**Reply by SmartManoj** [reply]
all true

**Reply by frankie567** [reply]
Could you copy-paste one of the generated token, so I could check if there's something wrong there?

**Reply by SmartManoj** [reply]
eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiNyIsImF1ZCI6ImZhc3RhcGktdXNlcnM6cmVzZXQiLCJleHAiOjE2NTU0NDcwMDF9.GUwp3aAPIc41VENBuIh6Wd7eLCdpDN3gdTYpKqDVtHg

**Reply by frankie567** [reply]
All right. So it seems you changed the type of ID to be integer then... Did you put the `IntegerIdMixin` on the `UserManager` instead of the UUID one?

```py
class UserManager(IntegerIdMixin, BaseUserManager[User, int]):
    ...
```

Ref: https://fastapi-users.github.io/fastapi-users/10.0/configuration/user-manager/#the-id-parser-mixin

**Reply by SmartManoj** [reply]
Yes but in
```
class User(SQLAlchemyBaseUserTable[int], Base):
    id = Column(Integer, primary_key=True,  autoincrement=True)
```

```
class UserManager(UUIDIDMixin, BaseUserManager[User, uuid.UUID]):
```

Solved after reverted to UUID.
